.github/workflows/build-pr.yml

```diff
@@ -88,3 +95,68 @@ jobs:
           name: e2e-logs
           path: |
             logs-*.json        
+  
+  build_test_aks_e2e:
+    name: "Build and run e2e Test on AKS"
+    runs-on: ubuntu-22.04
+    timeout-minutes: 15
+    permissions:
+      contents: read
+      id-token: write
+    strategy:
+      matrix:
+        KUBERNETES_VERSION: ["1.24.6"]
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v3
+      - name: Set up Go 1.19
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+
+      - name: Bootstrap e2e
+        run: |
+          mkdir -p $GITHUB_WORKSPACE/bin
+          echo "$GITHUB_WORKSPACE/bin" >> $GITHUB_PATH
+          make e2e-bootstrap KUBERNETES_VERSION=${{ matrix.KUBERNETES_VERSION }}
+
+      - name: Az CLI login
+        uses: azure/login@v1
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Connect to AKS cluster
+        run: |
+          az aks get-credentials --resource-group ${{ secrets.AZURE_GROUP }} --name ${{ secrets.AKS_NAME }} 
+
+      - name: Log into ACR
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.ACR_TEST_REGISTRY }}
+          username: ${{ secrets.APPLICATION_CLIENT_ID }}
+          password: ${{ secrets.APPLICATION_CLIENT_SECRET }}
```

Comment:
If logged into the azure account, isn't it sufficient to just do `az acr login`?

Reply:
I don't think you need to store a secret for login. Also, these secrets will need to be available to forks which is not secure.

Reply:
Good call, I'll test with that.